tests/test_wasi.py

```diff
@@ -10,83 +9,56 @@
 TEST_BYTES = open(here + '/wasi.wasm', 'rb').read()
 
 def test_wasi_version():
-    assert issubclass(WasiVersion, IntEnum)
-    assert len(WasiVersion) == 3
-    assert WasiVersion.Snapshot0 == 1
-    assert WasiVersion.Snapshot1 == 2
-    assert WasiVersion.Latest == 3
+    assert issubclass(wasi.Version, IntEnum)
+    assert len(wasi.Version) == 3
+    assert wasi.Version.SNAPSHOT0 == 1
+    assert wasi.Version.SNAPSHOT1 == 2
+    assert wasi.Version.LATEST == 3
+
+def test_wasi_get_version():
+    assert wasi.get_version(Module(Store(), "(module)"), strict=True) == None
+    assert wasi.get_version(Module(Store(), TEST_BYTES), strict=True) == wasi.Version.SNAPSHOT1
+
+def test_wasi_state_builder():
+    state_builder = \
+        wasi.StateBuilder("test-program"). \
+            argument("--foo"). \
+            environments({"ABC": "DEF", "X": "YZ"}). \
```

Comment:
`environments` sounds a bit strange in this context, I'd expect environments to be a even more plural, as in:

Environment variable - `EnvVar`
Environment variables - `Vec<EnvVar>`
Environments - `Vec<Vec<EnvVar>>` (that is, it's multiple sets of environment variables)

Reply:
I don't have a strong opinion here. I just don't want to get abbrevations. Would you prefer `environment_variables`?

Reply:
I think it would be good to use the same naming as python subprocess.run which is `env`:

https://docs.python.org/3/library/subprocess.html#subprocess.run